google/cloud/security/scanner/audit/instance_network_interface_rules_engine.py

```diff
@@ -0,0 +1,222 @@
+        is_external_network = rule_def.get('is_external_network')
+
+        if (whitelist is None) or (project is None) or (network is None)\
+                or (is_external_network is None):
+            raise audit_errors.InvalidRulesSchemaError(
+                'Faulty rule {}'.format(rule_def.get('name')))
+
+        rule_def_resource = {'whitelist': whitelist,
+                             'project': escape_and_globify(project),
+                             'network': escape_and_globify(network),
+                             'is_external_network': is_external_network}
+
+        rule = Rule(rule_name=rule_def.get('name'),
+                    rule_index=rule_index,
+                    rules=rule_def_resource)
+
+        resource_rules = self.resource_rules_map.get(rule_index)
+        if not resource_rules:
+            self.resource_rules_map[rule_index] = rule
+
+    def get_resource_rules(self):
+        """Get all the resource rules."""
+        return self.resource_rules_map.values()
+
+
+class Rule(object):
+    """The rules class for instance_network_interface."""
+
+    def __init__(self, rule_name, rule_index, rules):
+        """Initialize.
+
+        Args:
+            rule_name: Name of the loaded rule
+            rule_index: The index of the rule from the  definitions
+            rules: The resources associated with the rules like the whitelist
+        """
+        self.rule_name = rule_name
+        self.rule_index = rule_index
+        self.rules = rules
+
+    def find_violations(self, instance_network_interface):
+        """Raise violation is the ip is not in the whitelist.
+
+        Args:
+            instance_network_interface: InstanceNetworkInterface obj
+        """
+        network_and_project = re.search(
+            'compute\/v1\/projects\/([^\/]*).*networks\/([^\/]*)',
+            instance_network_interface.network)
+        project = network_and_project.group(1)
+        network = network_and_project.group(2)
+        is_external_network = instance_network_interface.access_configs \
+            is not None
+        if not self.rules['whitelist'].get(project):
+            yield self.RuleViolation(
+                resource_type='instance',
+                rule_name=self.rule_name,
+                rule_index=self.rule_index,
+                violation_type='INSTANCE_NETWORK_INTERFACE_VIOLATION',
+                project=project,
+                network=network,
+                ip='None')
+        elif network not in self.rules['whitelist'].get(project) and \
+                is_external_network:
+            yield self.RuleViolation(
+                resource_type='instance',
+                rule_name=self.rule_name,
+                rule_index=self.rule_index,
+                violation_type='INSTANCE_NETWORK_INTERFACE_VIOLATION',
+                project=project,
+                network=network,
+                ip=instance_network_interface.access_configs[0].get('natIP'))
```

Comment:
Readability: Fix the line length on this please. You can wrap like this:

ip=(instance_network_interface
    .access_configs[0].get('netIP')))